test/config_parser_SUITE.erl

```diff
@@ -2979,19 +2996,34 @@ err_host_or_global(Config) ->
 parse_host_config(Config) ->
     parse(#{<<"host_config">> => [Config#{<<"host_type">> => ?HOST}]}).
 
-%% plug in 'hosts' as this option is mandatory, then parse, then remove the extra 'hosts'
-parse(M)
-  when not is_map_key(<<"general">>, M) ->
-    parse(M#{<<"general">> => #{<<"hosts">> => [?HOST]}});
-parse(M = #{<<"general">> := GenM})
-  when not is_map_key(<<"hosts">>, GenM) ->
-    parse(M#{<<"general">> => GenM#{<<"hosts">> => [?HOST]}});
-parse(M) ->
+
+parse(M0) ->
+    %% 'hosts' (or 'host_types') and `default_server_domain` options are mandatory.
+    %% this function does the following things:
+    %%   1) plugs that mandatory options with dummy values (if required).
+    %%   2) executes parsing.
+    %%   3) removes extra 'hosts'/'default_server_domain' config keys (but only if
+    %%      they have dummy values).
+    %% DummyDomainName value must be unique to avoid accidental config keys removal.
+    DummyDomainName = <<"dummy.domain.name">>,
+    M = maybe_insert_dummy_domain(M0, DummyDomainName),
```

Comment:
The idea of putting `?HOST` here was to create a valid config. Now the config is invalid in some tests as `dummy.domain.name` is not the same host as in `host_config` (see `eq_host_config`). So the tests now show (and expect) that an invalid config is parsed correctly... This might be confusing.

I'd rather keep a valid host name here and fix the issue later than have a wrong value set on purpose.

Reply:
`eq_host_config` explicitly uses `?HOST` value in both checks.

Reply:
The point is that after the change some test cases are parsing inconsistent configuration, e.g.

```
#{<<"hosts">> => [<<"dummy...">>], <<"host_config">> => [{<<"host_type">> => ?HOST, ....}]}
```

Such configuration does not make sense and I think that it should result in an error and should be checked in a test case.